x/ecocredit/server/msg_server.go

```diff
@@ -345,5 +367,50 @@ func retire(ctx types.Context, store sdk.KVStore, recipient string, batchDenom b
 		Retirer:    recipient,
 		BatchDenom: string(batchDenom),
 		Units:      math.DecimalString(retired),
+		Location:   location,
 	})
 }
+
+// validateLocation checks that the country and region conform to ISO 3166 and
+// the postal code is valid
+func validateLocation(location string) error {
+	country, region, postalCode, err := parseLocation(location)
+	if err != nil {
+		return err
+	}
+	
+	if _, err := iso3166.Decode(country, region, false); err != nil {
+		return err
+	}
+
+	if err := validatePostalCode(postalCode); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func parseLocation(location string) (string, string, string, error) {
+	strings := strings.Split(location, "-")
+	switch numStrings := len(strings); numStrings {
+	case 1:
+		return strings[0], "", "", nil
+	case 2:
+		return strings[0], strings[1], "", nil
+	case 3:
+		return strings[0], strings[1], strings[2], nil
+	default:
+		return "", "", "", fmt.Errorf("Location should have format <country-code>[-<region-code>[-<postal-code>]]")
```

Comment:
maybe we should wrap the error (for instance with `sdkerrors.ErrInvalidRequest`?)